### PR TITLE
Added Firefox ESR version 91.0esr.

### PR DESCRIPTION
**Pull request by jedieaston**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

------

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/24383)

### CODE REVIEW

**Review by ItzLevvie — CHANGES_REQUESTED** [review]
image

For me, it seems to have installed the x86 MSI rather than the x64 MSI? Maybe the missing `Scope` in x64 is the issue? I am running it in administrator mode.

File: manifests/m/Mozilla/FirefoxESR/91.0esr/Mozilla.FirefoxESR.locale.en-US.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+LicenseUrl: https://www.mozilla.org/en-US/MPL/2.0/
+Moniker: firefoxesr
+Tags:
+- browser
+- firefox
+- mozilla
+- esr
+PackageUrl: https://www.mozilla.org/en-US/firefox/enterprise/
+Publisher: Mozilla
+PackageName: Firefox ESR
```
Comment (ItzLevvie) [inline manifests/m/Mozilla/FirefoxESR/91.0esr/Mozilla.FirefoxESR.locale.en-US.yaml:14]:
```suggestion
PackageName: Mozilla Firefox ESR
```

File: manifests/m/Mozilla/FirefoxESR/91.0esr/Mozilla.FirefoxESR.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.0.0.schema.json
+PackageIdentifier: Mozilla.FirefoxESR
+PackageVersion: 91.0esr
```
Comment (ItzLevvie) [inline manifests/m/Mozilla/FirefoxESR/91.0esr/Mozilla.FirefoxESR.yaml:3]:
```suggestion
PackageVersion: 91.0
```

File: manifests/m/Mozilla/FirefoxESR/91.0esr/Mozilla.FirefoxESR.installer.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.0.0.schema.json
+PackageIdentifier: Mozilla.FirefoxESR
+PackageVersion: 91.0esr
```
Comment (ItzLevvie) [inline manifests/m/Mozilla/FirefoxESR/91.0esr/Mozilla.FirefoxESR.installer.yaml:3]:
```suggestion
PackageVersion: 91.0
```

File: manifests/m/Mozilla/FirefoxESR/91.0esr/Mozilla.FirefoxESR.locale.en-US.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultlocale.1.0.0.schema.json
+PackageIdentifier: Mozilla.FirefoxESR
+PackageVersion: 91.0esr
```
Comment (ItzLevvie) [inline manifests/m/Mozilla/FirefoxESR/91.0esr/Mozilla.FirefoxESR.locale.en-US.yaml:3]:
```suggestion
PackageVersion: 91.0
```

**Review by ItzLevvie — CHANGES_REQUESTED** [review]

**Comment by jedieaston** [comment]
> For me, it seems to have installed the x86 MSI rather than the x64 MSI? Maybe the missing Scope in x64 is the issue? I am running it in administrator mode.

I thought winget installed as admin by default. Let me investigate.

**Comment by OfficialEsco** [comment]
> > For me, it seems to have installed the x86 MSI rather than the x64 MSI? Maybe the missing Scope in x64 is the issue? I am running it in administrator mode.
> 
> I thought winget installed as admin by default for MSIs, but I guess it just lets the installer do what it wants. Let me investigate.

I actually came across this issue a few days ago, WinGet prioritize the User scope by default, in this case it seems like it prefers the Installer with a Scope?

> I thought winget installed as admin by default for MSIs

I'm guessing you mean `Scope: machine`? Because WinGet does not prompt for admin rights, that is the installer, it will Select the installer bases on the preferences -> start the installer -> if installer requires admin, prompt for admin -> continue install

**Comment by jedieaston** [comment]
> I'm guessing you mean `Scope: machine`

Yes, that's what I meant. But I guess for a lot of scenarios that doesn't make sense.

**Comment by jedieaston** [comment]
It turns out if you're missing `InstallerLocale` WinGet will try to get the installer that has that even if there is one available that matches your architecture. Which makes sense since you would probably want one that you could read over one that matched your architecture.

**Comment by OfficialEsco** [comment]
>>I'm guessing you mean Scope: machine
>
>Yes, that's what I meant. But I guess for a lot of scenarios that doesn't make sense.

Yeah, i think this might have to be a issue on Winget-cli as i also do not want EVERY program to install to current user by default, if preferred scope is not set in `winget settings` it should pick the first installer in the Manifest matching your Locale and Architecture (I add the scopes this way already)

**Comment by jedieaston** [comment]
I don't think the order of how the installers are listed in the manifest is preserved in the sqlite file winget uses, so I'm not sure how they'd go about doing that. (And I personally don't mind this behavior, I understand why you want it to work that way but it makes more sense to try and install with as little privileges needed as possible IMO).

**Review by OfficialEsco — APPROVED** [review]

**Pull request merged**